### PR TITLE
London Banner

### PR DESCRIPTION
**Pull request by corwintines**
## Description
Adds a BannerNotfication for the London Upgrade. This banner will have a countdown until launch (which estimated time is fetched from etherscan.io), and will say the upgrade has been released when the block has been mined (again, fetched from etherscan.io).

## Related Issue

https://github.com/ethereum/ethereum-org-website/issues/3522

### CODE REVIEW

File: src/components/ReleaseBanner.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// Libraries
+import React, { useEffect, useState } from "react"
+import Countdown, { zeroPad } from "react-countdown"
+import styled from "styled-components"
+
+// Components
+import BannerNotification from "./BannerNotification"
+import Emoji from "./Emoji"
+import Translation from "./Translation"
```
Comment (wackerow) [inline src/components/ReleaseBanner.js:9]:
```suggestion
// Components
import BannerNotification from "./BannerNotification"
import Emoji from "./Emoji"
import Translation from "./Translation"
import Link from "./Link"
```

File: src/components/ReleaseBanner.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  const renderer = ({ days, hours, minutes, seconds, completed }) => {
+    if (completed) {
+      return (
+        <StyledBannerNotification>
+          <StyledEmoji size={2} text=":tada:" />
+          <Translation id="london-upgrade-banner-released" />
+          <Span>
+            <a href="/history/#london">
+              <Translation id="learn-more" />
+            </a>
```
Comment (wackerow) [inline src/components/ReleaseBanner.js:63]:
```suggestion
            <Link to="/history/#london">
              <Translation id="learn-more" />
            </Link>
```

File: src/components/ReleaseBanner.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          <StyledEmoji size={2} text=":tada:" />
+          <Translation id="london-upgrade-banner" />
+          <Span>
+            {zeroPad(days, 2)}:{zeroPad(hours, 2)}:{zeroPad(minutes, 2)}:
+            {zeroPad(seconds, 2)}!
+          </Span>
+          <Span>
+            <a href="/history/#london">
+              <Translation id="learn-more" />
+            </a>
```
Comment (wackerow) [inline src/components/ReleaseBanner.js:79]:
```suggestion
            <Link to="/history/#london">
              <Translation id="learn-more" />
            </Link>
```

**Review by wackerow — CHANGES_REQUESTED** [review]
@corwintines Love this! 👍🏻

Left a few comments that should fix the broken link... We use the `Link` component for all the links on the page, which will then parse it out according to the language and also if it is internal, external, or a glossary link.

**Review by pettinarip — APPROVED** [review]

**Comment by minimalsm** [comment]
LGTM 😎! 

Just a note for future upgrades if we're looking to use a banner like this it might be worth looking the design on smaller devices (< 320px) as things get a bit squashed.

Screenshot 2021-08-02 at 14 02 30

**Review by minimalsm — APPROVED** [review]

**Comment by minimalsm** [comment]
Just re-read @samajammin's comments on this from Discord

> I'm open to adding a site-wide banner if we'd like (e.g. in Layout.js) but in that case we'd want to remove the Eth2 template banners (& wherever else we have active banners, like if a docs page is incomplete), so it might actually be a bit of work. Perhaps we just add it to the homepage?

We've added this everywhere. This is how it looks on the pages with existing banners. Should we move it home page only or go with it as is for the 2-3 days? @samajammin @wackerow 

Screenshot 2021-08-02 at 14 18 45

Screenshot 2021-08-02 at 14 21 25

**Comment by corwintines** [comment]
poap

**Pull request merged**